firedrake_ts/ts_solver.py

```diff
@@ -81,15 +81,17 @@ def __init__(
         is_form_consistent(self.is_linear, self.bcs)
         self.Jp_eq_J = Jp is None
 
-        self.u = u
+        self.u_restrict = u
         self.udot = udot
         self.tspan = tspan
         self.F = F
         self.G = G
         self.Jp = Jp
-        if not isinstance(self.u, function.Function):
+        self.restrict = False
```

Comment:
This seems unused. Maybe you forgot to include some other files that were changed?

Reply:
Yes! Sorry, that's my mistake. It's a forgotten piece from one of my duck-typing compatibility experiments. Fix is incoming...